ctapipe/image/time_next_neighbor_cleaning.py

```diff
@@ -0,0 +1,652 @@
+import numpy as np
+from numba import njit
+from scipy.spatial import cKDTree as KDTree
+from scipy.interpolate import interp1d
+from ..containers import EventType
+from functools import lru_cache
+from tqdm.auto import tqdm
+
+import astropy.units as u
+import warnings
+import json
+
+
+def _get_neighbors_of(pixel_id, indptr, indices):
+    first = indptr[pixel_id]
+    last = indptr[pixel_id + 1]
+    return indices[first:last]
+
+
+def _get_groups_2nn(indptr, indices):
+    n_pixels = len(indptr) - 1
+    combinations = []
+    for pixel_id in range(n_pixels):
+        neighbors = _get_neighbors_of(pixel_id, indptr, indices)
+        for neighbor in neighbors:
+            if neighbor > pixel_id:
+                combinations.append((pixel_id, neighbor))
+
+    return np.array(combinations, dtype=np.uint16)
+
+
+def _get_groups_3nn(groups_2nn, indptr, indices):
+    combinations = []
+    for group in groups_2nn:
+        p1, p2 = group
+        neighbors_p1 = _get_neighbors_of(p1, indptr, indices)
+        neighbors_p2 = _get_neighbors_of(p2, indptr, indices)
+        for neighbor in np.intersect1d(neighbors_p1, neighbors_p2):
+            if neighbor > p1:
```

Comment:
combine to `if neighbor > p1 and neighbor > p2`

Reply:
Are you sure it is correct here to request this? Why are there no pixel groups where this does not hold true?